parts/linux/cloud-init/artifacts/cse_install.sh

```diff
@@ -261,18 +288,46 @@ installKubeletKubectlAndKubeProxy() {
 
     if [ -n "${KUBEPROXY_URL}" ]; then
         #kubeproxy is a system addon that is dictated by control plane so it shouldn't block node provisioning
-        pullContainerImage "docker" ${KUBEPROXY_URL} &
+        pullContainerImage ${CLI_TOOL} ${KUBEPROXY_URL} &
     fi
 }
 
 pullContainerImage() {
     CLI_TOOL=$1
-    DOCKER_IMAGE_URL=$2
-    retrycmd_if_failure 60 1 1200 $CLI_TOOL pull $DOCKER_IMAGE_URL || exit $ERR_CONTAINER_IMG_PULL_TIMEOUT
+    CONTAINER_IMAGE_URL=$2
+    if [[ ${CLI_TOOL} == "ctr" ]]; then
+        retrycmd_if_failure 60 1 1200 ctr --namespace k8s.io image pull $CONTAINER_IMAGE_URL || exit $ERR_CONTAINER_IMG_PULL_TIMEOUT
+    else 
+        retrycmd_if_failure 60 1 1200 ${CLI_TOOL} pull $CONTAINER_IMAGE_URL || exit $ERR_CONTAINER_IMG_PULL_TIMEOUT
+    fi
+}
+
+removeContainerImage() {
```

Comment:
is this func used?

Reply:
removing

Reply:
only used during vhd builds. moving to install-dependencies.sh.